### PR TITLE
Ft update profile 

### PR DESCRIPTION
**Pull request by chrill-christella**
### what does this PR do?

update the profile of the user

### Description of the task to be completed

- generate the user profile whenever a user sign up or login 
- create user profile controller and route to update the profile of a buyer or a sender
- add necessary validations

### CODE REVIEW

**Review by sostene12 — CHANGES_REQUESTED** [review]
please consider the conflicts you are having, it would help if you solve them first and then rebase the base branch (develop)

**Review by patsicko — CHANGES_REQUESTED** [review]
- We are normalizing user model by creating user profiles with respect to their respective roles.
I can suggest you to rebase ft: roles permissions #20 to have the latest changes. It will help.
- CI.yml has been changed to run a script that undo all migrations after each test. The same changes have been applied to the test script in package.json.  Hopefully this will solve the issues of your tests that are not passing.

**Review by Shumba250 — APPROVED** [review]

**Review by Luciefifi — APPROVED** [review]

File: src/api-docs/buyer.docs.js
Comment (sostene12) [inline src/api-docs/buyer.docs.js:1]: you are sending the id of user, hence you are finding by email from req.body

```
static async getProfile (req, res, next) {
  try {
    const { userId } = req.params;
    const {dataValues} = await user.findOne({where:{email:req.body.email}});
    const profile = await profiles.findUserProfile(userId);
    const {...others}= dataValues;
    if (profile) {
      res.json({ message: req.t('Found'), data: {others,profile} });
    } else {

      const Message = req.t('notfound');
      res.json({ message: Message });
    }
  } catch (error) {
    res.status(500).json({ error: error.message });
    next(error);
  }
```

File: src/routes/buyer.route.js
Comment (sostene12) [inline src/routes/buyer.route.js:1]: it can be good , if before updating you must login first

**Review by sostene12 — CHANGES_REQUESTED** [review]

**Comment by sostene12** [comment]
also, solve conflicts, and apply the comments on the vendor aswell

File: src/routes/buyer.route.js
Comment (chrill-christella) [reply src/routes/buyer.route.js:1]: resolved this issue too

**Review by wakaflorien — APPROVED** [review]

**Pull request merged**